### PR TITLE
feat: hook useCreateMultiShipOrderMutation

### PR DESCRIPTION
**Pull request by kibo-amolpawar**

### CODE REVIEW

File: lib/helpers/buildCreateCheckoutParams.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { Checkout } from '../gql/types'
```
Comment (TriptiVermaXP) [inline lib/helpers/buildCreateCheckoutParams.ts:1]: use type

File: hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  const queryClient = useQueryClient()
+
+  return useMutation(createOrder, {
+    onSuccess: () => {
+      queryClient.removeQueries([checkoutKeys.all])
+    },
+  })
+}
+
+export const useCreateMultiShipOrderMutation = useCreateCheckoutActionMutation
```
Comment (TriptiVermaXP) [inline hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx:38]: any specific reason behind reassigning useCreateCheckoutActionMutation to useCreateMultiShipOrderMutation ?
Can't we use useCreateCheckoutActionMutation for consistency with other files and also its already inside multiship folder?

**Review by TriptiVermaXP — APPROVED** [review]
Approving, please check few comments.

File: lib/helpers/buildCreateCheckoutParams.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+
+export const buildCreateCheckoutParams = (checkout: Checkout) => ({
```
Comment (kibo-kevinwatts) [inline lib/helpers/buildCreateCheckoutParams.ts:3]: I would suggest deleting this helper file and just creating arguments in the hook.

If you want to keep it though.
1. Change name so its related to "action". `buildCheckoutActionParams`
2. Function 'buildCheckoutActionParams' should take "actionName" as a parameter

File: hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { useMutation, useQueryClient } from 'react-query'
+
+import { makeGraphQLClient } from '@/lib/gql/client'
+import { createCheckoutActionMutation } from '@/lib/gql/mutations'
+import { buildCreateCheckoutParams } from '@/lib/helpers'
+import { checkoutKeys } from '@/lib/react-query/queryKeys'
+
+import type { Checkout, CheckoutActionInput } from '@/lib/gql/types'
+
+export interface OrderMultiShipInfo {
```
Comment (kibo-kevinwatts) [inline hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx:10]: rename `OrderMultiShipInfo` to `MultiShipCreateActionParams` or something "action" related

**Review by kibo-kevinwatts — CHANGES_REQUESTED** [review]

File: hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx (hunk 2 above)
Comment (kibo-amolpawar) [reply hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx:38]: In standard checkout we have  the same "useCreateCheckoutActionMutation" so for preventing conflict & to keep the same structure

File: hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx (hunk 4 above)
Comment (kibo-amolpawar) [reply hooks/mutations/multiShip/useCreateCheckoutActionMutation/useCreateCheckoutActionMutation.tsx:10]: Resolved. Renamed

File: lib/helpers/buildCreateCheckoutParams.ts (hunk 3 above)
Comment (kibo-amolpawar) [reply lib/helpers/buildCreateCheckoutParams.ts:3]: Resolved. Deleted

File: lib/helpers/buildCreateCheckoutParams.ts (hunk 1 above)
Comment (kibo-amolpawar) [reply lib/helpers/buildCreateCheckoutParams.ts:1]: Resolved. File deleted. not in use now.

**Review by kibo-sushant — APPROVED** [review]

**Review by kibo-kevinwatts — APPROVED** [review]

**Pull request merged**